compiler/daml-extension/src/extension.ts

```diff
@@ -99,34 +99,19 @@ function getViewColumnForShowResource(): ViewColumn {
     }
 }
 
-async function visualize() {
-    const util = require('util');
-    const exec = util.promisify(require('child_process').exec);
-    tmp.file(((err, path) => {
-        if (err) throw err;
-        let buildCmd = "daml build -o " + path
-        let visualizeCmd = "daml damlc visual " + path
-        let workspaceRoot = vscode.workspace.rootPath;
-        let execOpts = { cwd: workspaceRoot }
-        exec(buildCmd, execOpts, ((error: Error, stdout: string, stderr: string) => {
-            if (error) {
-                vscode.window.showErrorMessage("daml build failed with" + error)
-            }
-            else {
-                exec(visualizeCmd, execOpts, ((error: Error, stdout: string, stderr: string) => {
-                    if (error) {
-                        vscode.window.showErrorMessage("damlc visual command failed with " + error)
-                    }
-                    else {
-                        vscode.workspace.openTextDocument({ content: stdout, language: "dot" })
-                            .then(doc => vscode.window.showTextDocument(doc, vscode.ViewColumn.One, true)
-                                .then(_ => loadPreviewIfAvailable()))
-                    }
-                }))
-            }
-        }))
-
-    }))
+function visualize() {
+    if (vscode.window.activeTextEditor){
```

Comment:
What happens if I have an open window but it’s not a .daml file?

Reply:
Now should display an info message to open a DAML file